src/System.Collections.Specialized/src/System/Collections/Specialized/ListDictionary.cs

```diff
@@ -43,7 +43,7 @@ public ListDictionary(IComparer comparer)
                     while (node != null)
                     {
                         object oldKey = node.key;
-                        if (oldKey != null && oldKey.Equals(key))
+                        if (oldKey.Equals(key))
```

Comment:
What guarantees that ```oldKey``` is not null?  Is this because nulls can't be put into the collection as keys to begin with?